plugins/stringlib/stringlib.cpp

```diff
@@ -1580,6 +1580,11 @@ STRINGLIB_API unsigned STRINGLIB_CALL slMatchTimeOfDay(size32_t lenS, const char
     return 0;
 }
 
+#ifdef __GNUC__
```

Comment:
I tried using __attribute__((format)) but couldn't get it to work.

Reply:
I use 
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wformat-nonliteral"
 strftime ...
#pragma GCC diagnostic pop

Don't know other way except disable format-literal checking globally.  Yes attribute ((format)) doesn't work for strftime.